pkg/ds/daemon_set_test.go

```diff
@@ -309,7 +307,7 @@ func TestSchedule(t *testing.T) {
 	labeled = labeledPods(t, ds)
 	Assert(t).AreEqual(len(labeled), expectedNodes, "Expected no nodes to be unlabeled")
 
-	err = waitForPodsInIntent(consulStore, len(beforeDeletePods))
+	err = waitForPodsInIntent(consulStore, 12)
```

Comment:
> 12 is currently the right number because the 11 nodeQuality=good nodes on which the pod was previously scheduled hit the "Reality for this node matches this DS. No action required." and therefore they don't get new pods in intent.

shouldn't there be intent entries left over from one of the previous replications?